src/EventStore.Core/Certificates/CertificateUtils.cs

```diff
@@ -96,18 +97,36 @@ private static bool TryReadPkcs12CertificateBundle
 				}
 
 				// if the above attempt failed, try to load the certificate via the standard constructor
-				var certificate = new X509Certificate2(certificatePath, password);
+				var certificate = new X509Certificate2(certificatePath, certificatePassword);
 				if (!certificate.HasPrivateKey)
 					throw new NoCertificatePrivateKeyException();
 				return (certificate, null);
 			}
 
-			var privateKey = Convert.FromBase64String(string.Join(string.Empty, File.ReadAllLines(privateKeyPath)
-				.Skip(1)
-				.SkipLast(1)));
+			string[] allLines = File.ReadAllLines(privateKeyPath);
+			var header = allLines[0].Replace("-", "");
+			var privateKey = Convert.FromBase64String(string.Join(string.Empty, allLines.Skip(1).SkipLast(1)));
 
 			using var rsa = RSA.Create();
-			rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+			switch (header) {
+				case "BEGIN PRIVATE KEY":
+					rsa.ImportPkcs8PrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN RSA PRIVATE KEY":
+					rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN ENCRYPTED PRIVATE KEY":
+					if (privateKeyPassword is null) {
+						throw new ArgumentException(
+							"Need encryption password to read encrypted node private key");
+					}
+					rsa.ImportEncryptedPkcs8PrivateKey(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(privateKeyPassword)), new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				default:
+					throw new InvalidFileException("Node private key file format is unrecognizable");
```

Comment:
```suggestion
					throw new NotSupportedException($"Unsupported private key file format: {header}");
```

suggestion: throw a NotSupportedException instead of InvalidFileException

Reply:
Yes. done.